python/test_infra/test_utils.py

```diff
@@ -287,152 +283,78 @@ def wrapper(*args, **kwargs):
                 with open(output_file_name, "w") as f:
                     f.write(str(module))
 
-            return module if return_module else output_file_name
+            return module
 
         return wrapper
 
     return decorator
 
 
-def ttmetal_to_flatbuffer(
-    output_file_name: str = "ttmetal_fb.ttmg", golden_info: Dict[Operand, Golden] = None
+def ttnn_to_flatbuffer(
+    output_file_name: str = "ttnn_fb.ttnn",
 ):
     """
-    NOTE NOT WORKING, DO NOT USE.
-
-    Converts TTMetal module to flatbuffer and saves to file, meant to be used as a
-    decorator on top of `ttir_to_ttmetal` decorator. Take note that `ttir_to_ttmetal`
+    Converts TTNN module to flatbuffer and saves to file, meant to be used as a
+    decorator on top of `ttir_to_ttnn` decorator. Take note that `ttir_to_ttnn`
     has to return module instead of file name if decorated with this decorator.
 
-    Wrapper around `ttmetal_to_flatbuffer_file` pybound pass.
-
-    TODO Optional golden info is passed to be embedded in flatbuffer as well.
-
-    TODO Decorating a test function with this, i.e. calling
-    `ttmetal_to_flatbuffer_file` will result in
-
-    'LLVM ERROR: Building op `emitc.constant` but it isn't known in this MLIRContext:
-    the dialect may not be loaded or this operation hasn't been added by the dialect.'
-
-    To circumvent this, `ttmlir-translate` is run on file that
-    `ttir_to_ttmetal_backend_pipeline` produces to generate TTMetal flatbuffer file,
-    which this decorator was supposed to generate. Use `translate_ttmetal_to_flatbuffer`
-    to achieve this, and make `ttir_to_ttmetal` return file name instead of module.
+    Wrapper around `ttnn_to_flatbuffer_file` pybound pass.
     """
 
     def decorator(test_fn: Callable):
         def wrapper(*args, **kwargs):
-            # Get the TTMetal module by calling the wrapped function.
-            module = test_fn(*args, **kwargs)
+            # Get the TTNN module by calling the wrapped function.
+            module, builder = test_fn(*args, **kwargs)
 
             assert isinstance(module, Module), (
-                f"Make sure `ttir_to_ttmetal` which was decorated with this function "
+                f"Make sure `ttir_to_ttnn` which was decorated with this function "
                 f"returns module, not file name."
             )
 
             # Convert to flatbuffer file.
-            ttmetal_to_flatbuffer_file(module, output_file_name)
+            golden_info = {}
+            for name, golden_tensor in builder.id_golden_map.items():
+                golden_info[name] = create_golden_tensor(
+                    name,
+                    list(golden_tensor.tensor.shape),
+                    list(golden_tensor.tensor.stride()),
+                    DataType.Float32,
+                    golden_tensor.tensor.data_ptr(),
+                )
+            ttnn_to_flatbuffer_file(module, output_file_name, golden_info)
```

Comment:
You can neatly convert this into a public TTIRBuilder method `get_golden_map` if it is used in `ttmetal_to_flatbuffer` as well.

Reply:
> this golden struct wouldn't be equivalent for metal 

I'm confused by this statement, why wouldn't the golden struct be equivalent for metal?

Reply:
I guess I wanted to do some clever matching between ttir op and ttmetal decomposed ops. The only difference would be `golden_info[name]`. But I should correct myself: metal golden dump does work right now with the same flow that ttnn does. I added the same support from ttnn into ttmetal into this area as well.

Reply:
Cool sounds good, let's figure out clever matching separately